# 02 - Randomised Experiments

## The Golden Standard

1장에서는 상관관계와 인과관계의 차이점을 알아보고 상관관계를 인과관계로 만드는 방법을 공부했습니다.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$

편향이 없다면 ($E[Y_0|T=0]=E[Y_0|T=1]$) 상관관계는 인과관계가 됩니다. 실험군(처치가 적용됨)과 대조군(처치가 적용되지 않음)이 서로 같거나 비교할 수 있다면 상관관계는 인과관계가 됩니다. `untreated outcome`(처치되지 않은 결과)과 `treated outcome`(처치된 결과)의 `counterfactual outcome`(반사실적 결과)이 같을 때 비교할 수 있습니다. '반사실적 결과'는 '처치된 그룹이 처치를 받지 못했을 경우 얻었을 가상의 결과'라는 점을 기억하세요.

여기서는 편향을 없애기 위한 첫 번째 도구인 **Randomized Experiments**(무작위 실험)을 배웁니다. 무작위 실험에서는 데이터를 실험군과 대조군에 무작위로 할당합니다. 실험군의 비율이 50% 일 필요는 없습니다. 원한다면 실험군을 전체 중 10%로 할 수도 있습니다.

무작위화는 `potential outcome`(잠재 결과)을 `treatment`(처치)와 독립적으로 만들어 줍니다.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

처음에는 어려울 수 있어요. (저 또한 그랬습니다) 걱정하지 마시오. 나의 진정한 친구. 자세히 설명하겠소. 만약 결과와 처치가 독립이면, 처치의 효과가 없다는 것을 의미하지 않나요? 맞습니다! 하지만 우리는 결과에 대해 말하지 않습니다. **잠재 결과**에 대해 이야기하고 있어요. 잠재 결과는 **실험군($Y_1$)과 대조군($Y_0$) 내 데이터로 예상한 결과**입니다 (실험군과 대조군 각각에서 결과 예상치). 우기는 무작위 실험에서 **처치가 결과를 바꾼다고** 가정하므로 때문에 처치와 결과가 **독립적일 것을 원하지 않습니다.** 하지만 **잠재 결과**는 처치와 독립이어야 합니다.

<figure>
    <img src="data/img/rct/indep.png">
    <figcaption>(위) 잠재 결과와 독립 (아래)결과와 독립</figcaption>
</figure>

잠재 결과가 처치와 독립이면 실험군과 대조군이 같아집니다. 쉽게 말해 실험군과 대조군을 비교할 수 있습니다. 처치 할당 여부는 처치 전 결과에 대한 어떠한 정보도 주지 않습니다. 결과적으로 $(Y_0, Y_1)\perp T$는 오직 처치만이 실험군과 대조군 사이의 차이라는 것을 말해줍니다. 아래 수식은 독립성을 의미합니다.

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

1장에서 본 것처럼 다음 수식을 유도할 수 있습니다.

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0] = ATE
$

무작위화는 실험군과 대조군 사이의 평균 차이를 사용하는 방법을 제공하는데, 이러한 평균 차이를 `treatment effect`(처치 효과)라 합니다.


## In a School Far, Far Away

2020년 COVID-19는 사회적 거리두기를 강요했습니다. 배달 서비스가 널리 보급되고 대기업은 재택근무로 전환했습니다. 학교도 다르지 않습니다. 많은 학교가 온라인 강의를 시작했습니다.

위기가 발생한 지 4개월 지난 지금, 사람들은 이러한 변화가 계속될지 궁금해합니다. 온라인 학습의 이점은 의심할 여지가 없습니다. 온라인 학습은 월세나 교통비를 절약할 수 있어 저렴합니다. 또한, 세계적 수준의 콘텐츠를 더 디지털화할 수 있습니다. 여기서는 온라인 학습이 학생들의 학업 성취에 어떤 영향을 미치는지 알아봅시다. 

먼저 온라인 수업을 듣는 학생의 성적과 전통적인 강의를 듣는 학생의 성적을 비교할 수 있습니다. 이는 최선의 접근법이 아닙니다. 예를 들어 온라인 학교에는 이미 성적이 좋은 학생만 받을 가능성이 있습니다. 이 경우 처치(온라인 강의)에 대해 양의 편향이 있을 것입니다: $E[Y_0|T=1] > E[Y_0|T=0]$.

반면, 저렴한 비용 덕분에 다른 일을 하는 사람들이 주로 온라인 수업을 들을 수 있습니다. 이 경우 온라인 수업을 듣는 학생들의 성적이 평균적으로 나쁠 것이고 처치(온라인 강의)에 대해 음의 편향이 있을 것입니다: $E[Y_0|T=1] < E[Y_0|T=0]$.

간단한 비교는 설득력이 없습니다. 인과 효과를 감추는 편향이 있는지 확신할 수 없습니다.

![img](./data/img/rct/lurking_bias.png)

이를 해결하기 위해서는 비교할 수 있는 실험군과 대조군이 필요합니다 ($E[Y_0|T=1] = E[Y_0|T=0]$). 한 가지 방법은 학생들에게 온라인 수업을 무작위로 할당하는 것입니다. 무작위 할당으로 처치 외에는 학생들의 수준은 평균적으로 같아질 것입니다.

이에 대한 몇몇 연구 결과가 있습니다. 연구에서는 무작위로 학생들을 대면 수업, 온라인 수업, 대면/온라인 혼합 수업에 할당하고 학기 말 시험을 통해 성적을 확인했습니다.

데이터는 다음과 같습니다.

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv("./data/online_classroom.csv")
print(data.shape)
data.head()

(323, 10)


gender  asian  black  hawaiian  hispanic  unknown  white  format_ol  \
0       0    0.0    0.0       0.0       0.0      0.0    1.0          0   
1       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
2       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
3       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
4       1    0.0    0.0       0.0       0.0      0.0    1.0          1   

   format_blended  falsexam  
0             0.0  63.29997  
1             0.0  79.96000  
2             1.0  83.37000  
3             1.0  90.01994  
4             0.0  83.30000

데이터는 총 323개로 그다지 크진 않지만 충분합니다. 인과 관계를 추정하기 위해 각 그룹의 평균 점수를 계산할 수 있습니다.

In [2]:
(data
 .assign(class_format = np.select(
     [data["format_ol"].astype(bool), data["format_blended"].astype(bool)],
     ["online", "blended"],
     default="face_to_face"
 ))
 .groupby(["class_format"])
 .mean())

gender     asian     black  hawaiian  hispanic   unknown  \
class_format                                                               
blended       0.550459  0.217949  0.102564  0.025641  0.012821  0.012821   
face_to_face  0.633333  0.202020  0.070707  0.000000  0.010101  0.000000   
online        0.542553  0.228571  0.028571  0.014286  0.028571  0.000000   

                 white  format_ol  format_blended   falsexam  
class_format                                                  
blended       0.628205        0.0             1.0  77.093731  
face_to_face  0.717172        0.0             0.0  78.547485  
online        0.700000        1.0             0.0  73.635263

간단하죠? 대면 수업은 평균 점수가 78.54점, 온라인 수업은 73.63점입니다. 처치(온라인 강의)에 대한 $ATE$는 $-4.91$입니다. 따라서 **온라인 수업은 대면 수업에 비해 성적을 평균적으로 약 5점 낮게 만듭니다.** 무작위 실험은 처치 외 차이를 제거하기 위해 설계되었으므로 온라인 강의가 대면 수업을 할 수 없는 가난한 학생들로 구성되어 있다고 걱정할 필요가 없습니다. 온라인 강의를 받은 학생은 처치 외 어떤 면에서도 다르지 않습니다.

무작위화가 제대로 수행되었는지 (데이터가 올바른지) 확인하는 방법은 처치가 적용된 된 데이터가 그렇지 않은 데이터와 같은지 확인하는 것입니다. 데이터는 성별과 인종 정보를 가지고 있으므로 그룹 간 유사도 확인할 수 있습니다. `gender`, `asian`, `hispanic`, `white` 변수는 상당히 유사해 보입니다. 하지만 `black`은 조금 차이가 있어 보입니다. 무작위화하더라도 작은 데이터 세트에서는 우연히 한 그룹이 다른 그룹과 다를 수 있습니다. 큰 데이터 세트에서는 이러한 문제가 줄어듭니다.

## The Ideal Experiment

`Randomised experiments`(무작위 실험)과 `Randomised Controlled Trials`(RCT, 무작위 임상실험)은 인과 효과를 측정할 수 있는 가장 신뢰할 방법입니다. 매우 직접적인 기술로 너무나 설득력 있습니다. 대부분 국가에서 신약 효과를 증명하기 위한 요구사항으로 채택되고 있습니다. 할 수만 있다면 RCT는 인과 관계를 밝혀내기 위한 가장 강력한 도구입니다. 잘 설계된 RCT는 모든 과학자의 꿈입니다.

![img](./data/img/rct/science_dream.png)

하지만 RCT는 비용이 많이 들 수 있고 심지어 비윤리적일 수 있습니다. 단순히 할 수 없는 경우도 있습니다. 임신 중 흡연이 아기 체중에 미치는 영향을 알고 싶은 의사를 상상해 봅시다. 의사는 무작위로 임신 중인 엄마들에게 담배를 피우라고 할 수 없습니다. 큰 은행에서 일하는 직원이 신용 한도가 고객 이탈에 미치는 영향을 알고 싶다고 해봅시다. 고객에게 무작위로 신용한도를 설정하는 것은 너무나 큰 비용이 들 것입니다. 최저임금 인상이 실업률에 미치는 영향을 이해하고 싶나요? 국가에 무작위로 최저임금을 할당하라 할 수는 없습니다.

비윤리적이나 실현 불가능한 실험에 대해서는 할 수 있는 것이 없습니다. 그래도 인과적 질문에 대한 **이상적인 실험**을 생각해 볼 가치가 있습니다. **인과 효과를 밝히기 위해 할 수 있는 완벽한 실험이 무엇인지 항상 생각해 보세요.** 실제로 실험하지 않더라도 인과 효과를 발견할 방법을 알려줄지도 모릅니다.

## The Assignment Mechanism

무작위 실험에서 처치를 할당하는 방법은 무작위입니다. 앞으로 등장할 모든 인과 추론 기법은 처치의 할당 방법을 밝혀내려 할 것입니다. 처치 할당 방법을 알게 된다면, 꼭 무작위가 아닌 상황에서도 인과 추론에 대한 자신감이 붙을 겁니다.

할당 메커니즘은 데이터만으로는 찾을 수 없습니다. 예를 들어 교육과 자산에 대한 데이터 세트를 가지고 있을 때 데이터만으로는 어떤 것이 원인인지 알 수 없습니다. 당신은 세상에 대한 배경지식으로 그럴듯한 방법을 주장해야 합니다. 학교는 사람들을 교육하고, 더 생산적으로 만들고, 더 높은 임금을 받는 직업으로 이끌어 줄까요? 교육에 대해 비판적인 사람이라면 학교는 아무것도 하지 않는다고 주장할 수 있습니다. 단지 부유한 가정의 아이가 더 많은 교육을 받는 거짓된 상관관계이기 때문입니다.

X(교육)가 Y(자산)의 원인이라는 주장에 대해 X와 Y에 모두 영향을 주는 세 번째 변수 Z(부유한 가정)가 존재하므로 X와 Y간 관계가 거짓이라 주장할 수 있습니다. 이렇게 할당 매커니즘을 이해하는 것은 훨씬 설득력 있는 설명으로 이어집니다. 따라서 인과 추론은 재미있습니다. `ML`은 단순히 적절한 순서로 버튼을 누르는 것에 불과하지만, 인과 추론은 데이터 생성 방법을 진지하게 고민해야 합니다.

## Key Ideas

인과 관계를 발견하는 가장 간단하고 효과적인 방법인 무작위 실험을 공부했습니다. 무작위 실험은 실험군과 대조군을 비교할 수 있게 만들어 줍니다. 무작위 실험이 항상 가능하진 않지만, 이상적인 실험을 생각하는 것은 도움 될 수 있습니다.

통계를 알고 있는 사람이라면 추정치에 대한 평가가 없다고 지적할 수 있어요. $4.91$ 점 하락은 단지 우연은 아닐까요? 통계적으로 유의한가요? 타당한 의견입니다. 3장에서는 몇 가지 통계적 개념을 공부하겠습니다.

## References

I like to think of this entire book as a tribute to Joshua Angrist, Alberto Abadie and Christopher Walters for their amazing Econometrics class. Most of the ideas here are taken from their classes at the American Economic Association. Watching them is what is keeping me sane during this tough year of 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

I'll also like to reference the amazing books from Angrist. They have shown me that Econometrics, or 'Metrics as they call it, is not only extremely useful but also profoundly fun.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

My final reference is Miguel Hernan and Jamie Robins' book. It has been my trustworthy companion in the most thorny causal questions I had to answer.

* [Causal Inference Book](https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/)

The data used here is from a study of Alpert, William T., Kenneth A. Couch, and Oskar R. Harmon. 2016. ["A Randomized Assessment of Online Learning"](https://www.aeaweb.org/articles?id=10.1257/aer.p20161057). American Economic Review, 106 (5): 378-82.

![img](./data/img/poetry.png)

## Contribute

Causal Inference for the Brave and True is an open-source material on causal inference, the statistics of science. Its goal is to be accessible monetarily and intellectually. It uses only free software based on Python.
If you found this book valuable and want to support it, please go to [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). If you are not ready to contribute financially, you can also help by fixing typos, suggesting edits, or giving feedback on passages you didn't understand. Go to the book's repository and [open an issue](https://github.com/matheusfacure/python-causality-handbook/issues). Finally, if you liked this content, please share it with others who might find it helpful and give it a [star on GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).